tests/views/test_collection_exercise.py

```diff
@@ -1852,11 +1836,35 @@ def test_seft_upload_collection_instrument_supports_xls(self, mock_request, mock
         self.assertIn("Remove SEFT file".encode(), response.data)
 
     @requests_mock.mock()
+    @patch("response_operations_ui.controllers.collection_instrument_controllers.upload_collection_instrument")
+    @patch("response_operations_ui.views.collection_exercise.build_collection_exercise_details")
+    def test_seft_failed_upload_collection_instrument_with_error_message(
+        self, mock_request, mock_details, mock_upload_ci
+    ):
+        sign_in_with_permission(self, mock_request, user_permission_surveys_edit_json)
+        mock_upload_ci.return_value = (False, "Error message passed")
+        post_data = {"ciFile": (BytesIO(b"data"), "064_201803_0001.xlsx"), "load-ci": ""}
+        mock_request.get(url_get_survey_by_short_name, status_code=200, json=self.survey_data)
+        mock_request.get(url_ces_by_survey, json=self.collection_exercises)
+        mock_details.return_value = formatted_collection_exercise_details
+
+        response = self.client.post(
+            f"/surveys/{short_name}/{period}/load-collection-instruments", data=post_data, follow_redirects=True
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIn("Error: Failed to upload collection instrument".encode(), response.data)
+        self.assertIn("Error message passed".encode(), response.data)
+
+    @requests_mock.mock()
+    @patch("response_operations_ui.controllers.collection_instrument_controllers.upload_collection_instrument")
```

Comment:
The hierarchy in this test is not quite right. The responsibility of testing the different scenarios of the upload should be with test_collection_instrument_controller.py (which has been updated above) This test shouldn't stretch down that far as the combinations as you go further down into the code will only exponentially grow, we should only be concerned with the returns that come from upload_collection_instrument. With that in mind there was actually a missing test, which was added above